re-organize install instructions (#664)
carpentries/workshop-template: _extras/customization.md

title: Customizing Your Workshop's Website
permalink: /customization/index.html
---

## Table of Content

> * TOC

Reviewer: Do we need this `* TOC` line?

Reviewer: we do...

{:toc}

## Configuration File `_config.yml`

You should edit the `_config.yml` configuration file in the root directory of your workshop to
configure some site-wide variables and make the site function correctly:

> * `carpentry` - to tell us which carpentry workshop this is, possible values are ("swc" for Software
>   Carpentry workshops, "dc" for Data Carpentry workshops, "lc" for Library Carpentry Workshops, or
>   "cp" for general Carpentries events such as instructor trainings for which you should use
>   <https://github.com/carpentries/training-template> for your website template).
> * `curriculum` - for Data Carpentry and Software Carpentry, which one of the curriculum is being
>   taught. Possible values are: `dc-ecology`, `dc-genomics`, `dc-socsci`, `dc-geospatial`,
>   `swc-inflammation` and `swc-gapminder`.

Reviewer: ```suggestion
* `carpentry` - to tell us which carpentry workshop this is. Possible values are:
    - `"swc"` for Software Carpentry workshops,
    - `"dc"` for Data Carpentry workshops,
    - `"lc"` for Library Carpentry workshops, and
    - `"cp"` for general Carpentries events such as instructor trainings (for which you should use
      <https://github.com/carpentries/training-template> as the website template)
* `curriculum` - to tell us which curriculum is being taught.
  At the moment, applicable to Software and Data Carpentry workshops only.
  Possible values are:
    - `"dc-ecology"`, `"dc-genomics"`, `"dc-socsci"`, and `"dc-geospatial"` for Data Carpentry workshops
    - `"swc-inflammation"` and `"swc-gapminder"` for Software Carpentry workshops
```

Reviewer: I suggest using nested lists to show possible values. I also suggest adding double quotation marks around possible values to minimize possible confusion because things like `curriculum` and `carpentry` in `_config.yml` don't need those and possible values do.

> * `flavor` - `r` or `python` depending on which lessons are being taught at the workshop (currently

Reviewer: ```suggestion
* `flavor` - `"r"` or `"python"` depending on which lessons are being taught at the workshop (currently
```

title: Customizing Your Workshop's Website
permalink: /customization/index.html
---

## Table of Content

* TOC
{:toc}

## Configuration File `_config.yml`

You should edit the `_config.yml` configuration file in the root directory of your workshop to
configure some site-wide variables and make the site function correctly:

* `carpentry` - to tell us which carpentry workshop this is. Possible values are:
    - `"swc"` for Software Carpentry workshops,
    - `"dc"` for Data Carpentry workshops,
    - `"lc"` for Library Carpentry workshops, and
    - `"cp"` for general Carpentries events such as instructor trainings (for which you should use
      <https://github.com/carpentries/training-template> as the website template).

> 

Reviewer: it doesn't break anything but I don't think we need this empty line.

files are located in the appropriate workshop folder (`dc`, `lc` or `swc`)
inside the `_includes` folder.

## Home Page: Setup Instructions

If you need assistance with customizing the setup instructions for your website,
feel free to ask your questions in the Carpentries

> [Instructors Slack channel](https://swcarpentry.slack.com/archives/C08BVNU00)

Reviewer: noticed `swcarpentry` here. If desired, Slack org name can be changed to `carpentries`:
https://slack.com/help/articles/201663443-Change-your-workspace-or-org-name-and-URL

Author: yes, we have this in the works but there are some complications we need to resolve before we can finally move forward on this renaming.

1.  Add the workshop-template repository as upstream:

```bash
$ git remote add upstream https://github.com/carpentries/workshop-template.git
```

2.  Fetch the data from upstream repository (also know as the workshop-template
    repository):

```bash
$ git pull upstream
```

4.  Address possible merge conflicts, and

```bash
$ git commit -a
```

5.  Push the changes to your repository on GitHub:

```bash
$ git push origin gh-pages

> ```

Reviewer: Shouldn't these be standard Carpentries code blocks?
```
~~~
...
~~~
{: .language-bash}
```

I also suggest adding `{% include links.md %}` to the end of this document.

Author: :+1:

1.  Add the workshop-template repository as upstream:

~~~
$ git remote add upstream https://github.com/carpentries/workshop-template.git
~~~
{: .language-bash}

2.  Fetch the data from upstream repository (also know as the workshop-template
    repository):

~~~
$ git pull upstream
~~~
{: .language-bash}

4.  Address possible merge conflicts, and

~~~
$ git commit -a
~~~
{: .language-bash}

5.  Push the changes to your repository on GitHub:

~~~
$ git push origin gh-pages
~~~
{: .language-bash}

> 

Reviewer: Currently, this list appears as
```
1. ...

1. ...

1. ...
```

I submitted #674 against this branch to fix it.
